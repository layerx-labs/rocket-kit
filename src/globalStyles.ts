import { createGlobalStyle } from 'styled-components/macro';
import { rem} from "polished";
import { typography, colors } from './ions/variables';
import { device } from './ions/breakpoints';

const GlobalStyle = createGlobalStyle`
  * {
    box-sizing: border-box;
  }

  html {
    font-size: 16px;
    scroll-behavior: smooth;
  }

  body {
    margin: 0;
    font-family: 'Space Grotesk', Verdana, Arial, Helvetica, sans-serif;
    font-size: 1rem;
    font-weight: ${typography.regular};
    color: ${colors.black};
    font-feature-settings: 'ss04' on;
    -webkit-font-smoothing: antialiased;
  }

  h1,
  h1 span,
  h2,
  h2 span {
    margin: 0;
    font-weight: ${typography.regular};
  }

  h1,
  h1 span {
    font-family: 'Laforgestencil', Verdana, Arial, Helvetica, sans-serif;
    font-size: 3.5rem;
    line-height: 1.15;
  }

  h2,
  h2 span {
    font-size: 2rem;
    line-height: 1.2;
  }

  h3,
  h4 {
    margin: 0;
    font-size: 1rem;
    font-weight: 400;
  }

  span {
    font-family: ${typography.defaultFont};
    font-size: ${rem(typography.defaultSize)};
    font-feature-settings: 'ss04' on;
  }

  p {
    margin: 0;
    font-size: 1rem;
    line-height: 1.3;

    &:not(:last-child) {
      margin-bottom: ${rem('10px')};
    }
  }

  pre {
    border: 1px solid ${colors.grey500};
    border-radius: 4px;
    background-color: ${colors.grey300};
    padding: ${rem('15px')};
  }

  @media ${device.l} {
    h1,
    h1 span {
      font-size: 6rem;
      line-height: 1.1;
    }

    h2,
    h2 span {
      font-size: 2.5rem;
    }
  }
`;

export default GlobalStyle;
