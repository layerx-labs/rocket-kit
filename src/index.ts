// Test
export { Thing } from './thing';

// Ions
export { default as TaikaiLogo } from './ions/taikai-logo';

// Atoms
export { default as Avatar } from './atoms/avatar';
export { default as Checkbox } from './atoms/checkbox';
export { default as Tag } from './atoms/tag';
