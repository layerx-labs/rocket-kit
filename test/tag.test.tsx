import React from 'react';
import { render, screen } from '@testing-library/react';
import { Tag } from '../src';

describe('Tag', () => {
  it('renders', () => {
    const { asFragment } = render(<Tag color={'danger'} value={'hello'} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Tag has correct value', async () => {
    const tagValue = 'This is my tag';
    render(<Tag color={'danger'} value={tagValue} />);

    const foundText = await screen.getByText(tagValue);
    expect(foundText).toBeTruthy();
  });
});
