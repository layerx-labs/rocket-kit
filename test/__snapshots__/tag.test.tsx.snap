// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders 1`] = `
<DocumentFragment>
  <span
    class="sc-jSgupP iIWTSr"
    color="danger"
  >
    hello
  </span>
</DocumentFragment>
`;
